src/heap/src/SystemHeap.c

```diff
@@ -321,6 +322,11 @@ DEFINE_FOOTER_SIZE(sysGetAllocationFooterSize)
     return SYS_ALLOCATION_FOOTER_SIZE;
 }
 
+DEFINE_ALIGNED_SIZE(sysGetAllocationAlignedSize)
```

Comment:
Looks like the macro name is same for System Heap as well as Hybrid heap. Just for sake of clarity, can we rename these to indicate the type of heaps? `DEFINE_ALIGNED_SIZE_SYSTEM_HEAP` probably?

Reply:
These are helper macros that define the functions for various implementations. Currently, we have 3 implementations and all would use these macros to define the same functions. Then, in the Heap.c we implement the polymorphism - this is the pattern we use in multiple other places to achieve polymorphism without expensive v-tables.